### PR TITLE
Fix grab-site and python3Packages.lmdb

### PR DESCRIPTION
**Pull request by ivan**
###### Motivation for this change

This fixes:
- python3Packages.lmdb
  - upgraded lmdb to fix a failing test with lmdb 0.9.28: https://github.com/jnwatson/py-lmdb/issues/278
  - removed the strange dependency on ludios_wpull, which was breaking the python38Packages.lmdb build
- grab-site, which depends on python37Packages.lmdb

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc maintainers @copumpkin @jb55 @vcunat 

I confirmed that grab-site now builds and runs on NixOS master.

### CODE REVIEW

**Comment by vcunat** [comment]
The C part looks good to me; I'm not really familiar with the python package.

**Comment by lukegb** [comment]
@ofborg build python3.pkgs.lmdb

**Review by lukegb — APPROVED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120785` at 68756398 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>61 packages marked as broken and skipped:</summary>
  <ul>
    <li>gnome3.gnome-books</li>
    <li>gnome3.gnome-documents</li>
    <li>libsForQt512.baloo</li>
    <li>libsForQt512.baloo-widgets</li>
    <li>libsForQt512.discover</li>
    <li>libsForQt512.dolphin</li>
    <li>libsForQt512.dolphin-plugins</li>
    <li>libsForQt512.dragon</li>
    <li>libsForQt512.elisa</li>
    <li>libsForQt512.gwenview</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/5812deb25182870b4b733b4798f5d993">pantheon.appcenter</a></li>
  </ul>
</details>
<details>
  <summary>48 packages skipped due to time constraints:</summary>
  <ul>
    <li>caffe</li>
    <li>denemo</li>
    <li>dero</li>
    <li>gerbil</li>
    <li>gerbil-unstable</li>
    <li>glom</li>
    <li>kdev-php</li>
    <li>kdev-python</li>
    <li>kdevelop-unwrapped</li>
    <li>libmodsecurity</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>44 packages built successfully:</summary>
  <ul>
    <li>appstream</li>
    <li>bookworm</li>
    <li>evince (gnome3.evince)</li>
    <li>gawkextlib.lmdb</li>
    <li>gnome-feeds</li>
    <li>gnome3.sushi</li>
    <li>grab-site</li>
    <li>haskellPackages.lmdb</li>
    <li>haskellPackages.lmdb-high-level</li>
    <li>haskellPackages.lmdb-simple</li>
    <li>knot-dns</li>
    <li>knot-resolver</li>
    <li>libsForQt5.appstream-qt (libsForQt515.appstream-qt ,plasma5Packages.appstream-qt)</li>
    <li>libsForQt5.baloo (libsForQt515.baloo ,plasma5Packages.baloo)</li>
    <li>libsForQt5.baloo-widgets (libsForQt515.baloo-widgets ,plasma5Packages.baloo-widgets)</li>
    <li>libsForQt5.dragon (libsForQt515.dragon ,plasma5Packages.dragon)</li>
    <li>libsForQt5.kde-cli-tools (libsForQt515.kde-cli-tools ,plasma5Packages.kde-cli-tools)</li>
    <li>libsForQt5.khotkeys (libsForQt515.khotkeys ,plasma5Packages.khotkeys)</li>
    <li>libsForQt5.kmenuedit (libsForQt515.kmenuedit ,plasma5Packages.kmenuedit)</li>
    <li>libsForQt5.krohnkite (libsForQt515.krohnkite ,plasma5Packages.krohnkite)</li>
    <li>libsForQt5.kwin-dynamic-workspaces (libsForQt515.kwin-dynamic-workspaces ,plasma5Packages.kwin-dynamic-workspaces)</li>
    <li>libsForQt5.kwin-tiling (libsForQt515.kwin-tiling ,plasma5Packages.kwin-tiling)</li>
    <li>libsForQt5.parachute (libsForQt515.parachute ,plasma5Packages.parachute)</li>
    <li>libsForQt5.plasma-browser-integration (libsForQt515.plasma-browser-integration ,plasma5Packages.plasma-browser-integration)</li>
    <li>libsForQt5.plasma-disks (libsForQt515.plasma-disks ,plasma5Packages.plasma-disks)</li>
    <li>libsForQt5.plasma-workspace (libsForQt515.plasma-workspace ,plasma5Packages.plasma-workspace)</li>
    <li>libsForQt5.systemsettings (libsForQt515.systemsettings ,plasma5Packages.systemsettings)</li>
    <li>libsForQt514.appstream-qt</li>
    <li>libsForQt514.baloo</li>
    <li>libsForQt514.baloo-widgets</li>
    <li>lmdb</li>
    <li>metadata-cleaner</li>
    <li>meteo</li>
    <li>pantheon.elementary-calculator</li>
    <li>pantheon.elementary-camera</li>
    <li>pantheon.elementary-code</li>
    <li>pantheon.elementary-greeter</li>
    <li>pantheon.elementary-terminal</li>
    <li>pantheon.wingpanel-applications-menu</li>
    <li>pantheon.wingpanel-with-indicators</li>
    <li>python38Packages.lmdb</li>
    <li>python39Packages.lmdb</li>
    <li>setzer</li>
    <li>snippetpixie</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`checkPhase` should probably contain `runHook preCheck` and `runHook postCheck`.

Near pkgs/development/libraries/lmdb/default.nix:28:3:
```
   |
28 |   checkPhase = "make test";
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/libraries/lmdb/default.nix:15:15:
```
   |
15 |   patches = [ ./hardcoded-compiler.patch ];
   |               ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Comment by jb55** [comment]
On Mon, Apr 26, 2021 at 07:57:58PM +0000, Ivan Kozik wrote:
>This makes no sense and yet was added in 65eccfad5d32328d12cf8b42d02560905a8c9bae
>
>This was breaking the build of python3Packages.lmdb because ludios_wpull does not
>build on 3.8, only 3.7
>-  propogatedBuildInputs = [ ludios_wpull ];
>-

weird, good catch. ACK.

File: pkgs/development/libraries/lmdb/default.nix [hunk 1]
```diff
@@ -2,12 +2,12 @@
 
 stdenv.mkDerivation rec {
   pname = "lmdb";
-  version = "0.9.28";
+  version = "0.9.29";
 
   src = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/lmdb/default.nix:7]: This could be converted to fetchFromGitLab.

Comment (ivan) [reply pkgs/development/libraries/lmdb/default.nix:7]: Sure, but I hope that doesn't need to hold this up

Comment (SuperSandro2000) [reply pkgs/development/libraries/lmdb/default.nix:7]: Also please update the homepage to https://symas.com/lmdb/technical/

**Review by mweinelt — APPROVED** [review]

**Pull request merged**